processing/src/main/java/io/druid/query/IntervalChunkingQueryRunner.java

```diff
@@ -31,29 +34,38 @@
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ExecutorService;
 
 /**
  */
 public class IntervalChunkingQueryRunner<T> implements QueryRunner<T>
 {
   private final QueryRunner<T> baseRunner;
-  private final Period period;
 
-  public IntervalChunkingQueryRunner(QueryRunner<T> baseRunner, Period period)
+  private final QueryToolChest<T, Query<T>> toolChest;
+  private final ExecutorService executor;
+  private final QueryWatcher queryWatcher;
+  private final ServiceEmitter emitter;
+
+  public IntervalChunkingQueryRunner(QueryRunner<T> baseRunner, QueryToolChest<T, Query<T>> toolChest,
+      ExecutorService executor, QueryWatcher queryWatcher, ServiceEmitter emitter)
   {
     this.baseRunner = baseRunner;
-    this.period = period;
+    this.toolChest = toolChest;
+    this.executor = executor;
+    this.queryWatcher = queryWatcher;
+    this.emitter = emitter;
   }
 
   @Override
   public Sequence<T> run(final Query<T> query, final Map<String, Object> responseContext)
   {
-    if (period.getMillis() == 0) {
+    final Period p = getChunkPeriod(query);
```

Comment:
can we have a better naming convention than using a single letter? makes reading the rest of the code difficult